File: SonicXBackned/src/main/java/org/example/Server/User/RegisterHandler.java
package org.example.Server.User;


import com.sun.net.httpserver.HttpHandler;
import com.sun.net.httpserver.HttpExchange;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.node.ObjectNode;
import org.example.Server.RequestData;
import org.example.UserManagment.*;

import java.io.IOException;
import java.io.OutputStream;


public class RegisterHandler implements HttpHandler {
    @Override
    public void handle(HttpExchange exchange) throws IOException {
        if ("POST".equals(exchange.getRequestMethod())) {
            ObjectMapper objectMapper = new ObjectMapper();
            RequestData requestData = objectMapper.readValue(exchange.getRequestBody(), RequestData.class);

            System.out.println("Gelen veri: " + requestData);

            UserService userService = UserService.getInstance();
            boolean isSucces= userService.register(requestData.getFirstName(), requestData.getLastName(), requestData.getEmail(), requestData.getPassword());

            ObjectNode responseJson = objectMapper.createObjectNode();
            responseJson.put("success", isSucces);

            byte[] response = responseJson.toString().getBytes();
            exchange.getResponseHeaders().add("Content-Type", "application/json");
            exchange.sendResponseHeaders(200, response.length);
            OutputStream os = exchange.getResponseBody();
            os.write(response);
            os.close();
        } else {
            exchange.sendResponseHeaders(405, -1); // 405 Method Not Allowed
        }
    }
}

File: SonicXBackendC/SonicXBackendC/Server/User/RegisterHandler.cs
using Newtonsoft.Json;
using SonicXBackendC.UserManagement;
using System;
using System.IO;
using System.Net;

namespace SonicXBackendC.Server.User
{
    public class RegisterHandler
    {
        public void Handle(HttpListenerContext context)
        {
            if (context.Request.HttpMethod.Equals("POST", StringComparison.OrdinalIgnoreCase))
            {
                using (var reader = new StreamReader(context.Request.InputStream))
                {
                    var requestBody = reader.ReadToEnd();
                    var requestData = JsonConvert.DeserializeObject<RequestData>(requestBody);

                    Console.WriteLine("Gelen veri: " + requestData);

                    var userService = UserService.GetInstance();
                    var isSuccess = userService.Register(requestData.FirstName, requestData.LastName, requestData.Email, requestData.Password);
                    Console.WriteLine(isSuccess);
                    var responseJson = new { success = isSuccess };
                    var response = System.Text.Encoding.UTF8.GetBytes(JsonConvert.SerializeObject(responseJson));
                    context.Response.ContentType = "application/json";
                    context.Response.StatusCode = (int)HttpStatusCode.OK;
                    context.Response.OutputStream.Write(response, 0, response.Length);
                }

                context.Response.Close();
            }
            else
            {
                context.Response.StatusCode = (int)HttpStatusCode.MethodNotAllowed;
                context.Response.Close();
            }
        }
    }
}
